src/pages/index.js

```diff
@@ -41,38 +59,77 @@ function handleImageClick(name, link) {
 }
 
 //Popup Forms
-const profilePopup = new PopupWithForm(profileModal, (data) => {
-  console.log(data);
+const profilePopup = new PopupWithForm(profileModal, (data, button) => {
   userInfo.setUserInfo(data);
-  profilePopup.closePopup();
-});
-const cardForm = new PopupWithForm(newCardModal, (data) => {
-  const name = newCardTitle.value;
-  const link = newCardURL.value;
-  addCard({ name, link }, mediaList);
-  cardForm.closePopup();
+  api.saveUserInfo(data).then(() => {
+    button.textContent = "Save";
+    profilePopup.closePopup();
+  });
+  button.textContent = "Saving...";
 });
+
+const profilePicturePopup = new PopupWithForm(
+  profilePictureModal,
+  (data, button) => {
+    api
+      .updatePicture(data.url)
+      .then((res) => {
+        profileAvatar.src = data.url;
+        button.textContent = "Save";
+        profilePicturePopup.closePopup();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    button.textContent = "Saving...";
+  }
+);
+const deletePopup = new PopupWithFormSubmit(deleteModal);
+deletePopup.setEventListeners();
+
+function handleDeleteCard(cardId, cardElement) {
+  const id = cardId;
+  deletePopup.openPopup();
+  deletePopup.setSubmitAction(() => {
+    api.deleteCard(id);
+    cardElement.remove();
+    deletePopup.closePopup();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**